### DISCUSSION TITLE
WEBVTT format on SkyShowtime

### DISCUSSION THREAD

**Discussion by OtaStrom** [SubtitleEdit/subtitleedit#6944] [Ideas]
There is an extended WEBVTT file format used on SkyShowtime which hasn't been implemented yet.
Here are some examples:

[SkyShowtime_examples.zip](https://github.com/SubtitleEdit/subtitleedit/files/11502757/SkyShowtime_examples.zip)

For now, I use regular expressions to deal with these:

First the italic:
`<c\.[^>]*?italic.*?>((.*?)|(.*?\n.*?))</c[^>]*?>` --> `<i>$1</i>`

Then everything else with find and replace with nothing:
1 (start tag): `<c\.[^>]*?>`
2 (end tag): `</c[^>]*?>`

**Comment by GrampaWildWilly** [comment]
> ** Reply to message from OtaStrom ***@***.***> on Wed, 17 May 2023 15:01:38 -0700

This sounded intriguing to me so I did a little experimenting.  Since you can't really
open a VTT file in SE without a video, I copied one of my operas to a convenient place. 
I have lots of operas from the Metropolitan Opera that are a combination of video without
audio & audio without video, which play fine using VLC's synchronous playback feature. 
So I copied one of my video-only operas into the directory where I had unzipped your file
& renamed the .mp4 to match your .vtt.  Your subtitles are over an hour long so I needed
to pick something at least as long as your subtitles.  An opera seemed like an obvious
choice.  Then I tried to play the video in VLC, not in SE.  Of course, it was a silent
opera, something some people would say is an improvement.  Not I, but some people.  No
subtitles appeared.  This is VLC 3.0.18, the latest public release.  Clearly, they are
way behind the times on this.

Then I tried opening the .vtt in SE.  It was displaying all the tags in the MPV window. 
Worse, they were partially outside the window.  But I was using a woefully out of date
SE.  So I updated to SE 3.6.13 beta 8, which is the latest one, as far as I can tell.  It
displays the subtitles just fine.  I don't know if it is obeying the directives contained
in the tags that appear to be requesting certain colors.  Clearly, I'm not an extreme
expert in the WEBVTT format.  And the strings marked for italics are not being displayed
as italics.  But at least the subtitles are displayed.  That's a start.  I tried each of
your .vtt files in turn, renaming the opera .mp4 as necessary, & they all appeared to
display the subtitles.  Perhaps this will be good enough for the time being that you
don't have to remove the tags, so at least you'll still have them when Nik gets around to
supporting this WEBVTT syntax.

Or will he?  You say that this "is an extended WEBVTT file format used on SkyShowtime." 
Extended by whom?  By the international standards body that controls the WEBVTT format? 
Or by SkyShowtime?  If the former, it would be reasonable to expect Nik to react.  But if
it's some proprietary format chosen by the TV network, I would think Nik would ignore it.
Is this a new WEBVTT syntax?  How recent?

By the way, what language is that?  Czech?

WW
GO HABS GO!

**Reply by OtaStrom** [reply]
> Since you can't really open a VTT file in SE without a video ...

Yes, you can actually.


> You say that this "is an extended WEBVTT file format used on SkyShowtime." Extended by whom? By the international standards body that controls the WEBVTT format? Or by SkyShowtime? If the former, it would be reasonable to expect Nik to react. But if it's some proprietary format chosen by the TV network, I would think Nik would ignore it. Is this a new WEBVTT syntax? How recent? By the way, what language is that? Czech?

I'm not sure how crucial this information is to the problem. But anyway, I can't answer that. I am also not an expert on this format. But since I found it on SkyShowtime, I'd assume it's their "adjustment...
I'd prefer not to get into wordplay if I've stated something inaccurately somewhere. The bottom line is that there is this file type and I'm making it known here for consideration whether to include support for this type.

I don't know how you open the subtitles, but even in the latest Beta version, I don't see that the tags are automatically removed (which is good for me, because then I have the option to fix them).

And yes, it's Czech.

**Comment by GrampaWildWilly** [comment]
> ** Reply to message from OtaStrom ***@***.***> on Wed, 17 May 2023 23:08:35 -0700

Oh.  I didn't mean to imply that SE removes the tags, just that it doesn't display them
in the MPV video player.  (I tried to make the subtitles appear in VLC outside SE & they
didn't even display, not the tags, not the subtitles themselves.  I would expect that
would mean that in SE with VLC selected as the video player, it might have trouble as
well.  I didn't try that so I'm just guessing.)   I thought you would want to keep the
tags.  Why do you want to get rid of them?  I think they're supposed to provide certain
formatting.  Don't you want the formatting to be honored once you find a player that can
interpret them?

By "open a VTT" I meant double click the VTT.  That causes SE to open with that file
displayed.  That is, assuming you've made the relevant file associations in your system. 
You can also open a subtitle file in a text editor, like for example Notepad++.  It's
just plain text, after all.  For the kind of changes you're doing, I might be tempted to
do them in Notepad++ instead of SE.  But to each his own.

WW
GO HABS GO!

**Comment by GrampaWildWilly** [comment]
> ** Reply to message from OtaStrom ***@***.***> on Wed, 17 May 2023 23:08:35 -0700

As for opening a subtitle file in SE without a video, I never even thought of doing that.
I tried it just now.  It's kind of unusual.  Sure, you can edit the subtitles, but the
video player doesn't show them.  I even pressed the Play button in the MPV player & the
right triangle dutifully changed to || as if it were playing something, but the subtitles
didn't display.  I think I would always want the visual feedback from seeing the effect
of whatever I'm doing to the subtitles.

**Comment by niksedk** [comment]
Latest version (3.6.13) lets mpv handle the display of WebVTT, but it seems like mpv has a very basic viewing of WebVTT.

In SE, you can just Change the format to "SubRip (.srt)" and the WebVTT tags will be removed.

I think SE could implement some basic style reading, and make some basic WebVTT to ASSA converting, so mpv could do a better preview.
Better positioning support of WebVTT has also been requested.
I think this is a good idea, but I don't know when I will have the time...
